### PR TITLE
fix(wiki): sanitize db test connection

### PR DESCRIPTION
**Pull request by jeremyjaouen**
## Description

This PR fix a security breach into knowledge base form.

**Fixes** #4549 #4550

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [x] 2.8.x

## Checklist

- [x] I followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: www/api/class/centreon_wiki.class.php [hunk 1]
```diff
@@ -45,8 +45,11 @@ class CentreonWiki extends CentreonWebService
     public function postCheckConnection()
     {
         $sql_host = explode(':', $this->arguments['host']);
-        $host = $sql_host[0];
-        $port = isset($sql_host[1]) ? $sql_host[1] : '3306';
+
+        $host = filter_var($sql_host[0], FILTER_SANITIZE_STRING);
+        if (isset($sql_host[1])) {
```
Comment (loiclau) [inline www/api/class/centreon_wiki.class.php:50]: default value if $sql_host[1] not set?

File: www/api/class/centreon_wiki.class.php [hunk 2]
```diff
@@ -45,8 +45,11 @@ class CentreonWiki extends CentreonWebService
...
+            $port = filter_var($sql_host[1], FILTER_VALIDATE_INT) ? $sql_host[1] : '3306';
+        }
```
Comment (sc979) [inline www/api/class/centreon_wiki.class.php:52]:
```suggestion
        !empty($sql_host[1]) && null !== filter_var($sql_host[1], FILTER_VALIDATE_INT) ? $sql_host[1] : '3306';
```

File: www/api/class/centreon_wiki.class.php (hunk 1 above)
Comment (jeremyjaouen) [reply www/api/class/centreon_wiki.class.php]: solved with [last commit](https://github.com/centreon/centreon/pull/8875/commits/af803bb411c090c8f142d334db066cf0cf4868a5)

File: www/api/class/centreon_wiki.class.php (hunk 2 above)
Comment (jeremyjaouen) [reply www/api/class/centreon_wiki.class.php]: solved with [last commit](https://github.com/centreon/centreon/pull/8875/commits/af803bb411c090c8f142d334db066cf0cf4868a5)

**Pull request merged**